src/reorder.rs

```diff
@@ -198,6 +201,30 @@ fn group_imports(uts: Vec<UseTree>) -> Vec<Vec<UseTree>> {
     vec![std_imports, external_imports, local_imports]
 }
 
+/// Divides imports into two groups, corresponding to external crates
+/// and local imports. Sorts each subgroup.
+fn group_imports_external_crate(uts: Vec<UseTree>) -> Vec<Vec<UseTree>> {
+    let mut external_imports = Vec::new();
+    let mut local_imports = Vec::new();
+
+    for ut in uts.into_iter() {
+        if ut.path.is_empty() {
+            external_imports.push(ut);
+            continue;
+        }
+        match &ut.path[0].kind {
+            UseSegmentKind::Ident(..) => external_imports.push(ut),
+            UseSegmentKind::Slf(_) | UseSegmentKind::Super(_) | UseSegmentKind::Crate(_) => {
+                local_imports.push(ut)
+            }
+            // These are probably illegal here
+            UseSegmentKind::Glob | UseSegmentKind::List(_) => external_imports.push(ut),
+        }
+    }
+
+    vec![external_imports, local_imports]
+}
+
```

Comment:
There could be a way to deduplicate this and `StdExternalCrate`, by either calling `group_imports_std_external_crate` in this function or calling this function in the other function.

Reply:
It's been a while since I was in the code, but I have a vague recollection that there wasn't much that was useful to DRY up here. That being said, WDYT about something that results in:

```rust
    for ut in uts {
        match UseCategory::categorize(ut) {
            UseCategory::Std(ut) => std_imports.push(ut),
            UseCategory::External(ut) => external_imports.push(ut),
            UseCategory::Local(ut) => local_imports.push(ut),
        }
    }
```
and
```rust
    for ut in uts {
        match UseCategory::categorize(ut) {
            UseCategory::Std(ut) | UseCategory::External(ut) => external_imports.push(ut),
            UseCategory::Local(ut) => local_imports.push(ut),
        }
    }
```